src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1221,6 +1221,17 @@ public virtual bool IsTupleMethod
             }
         }
 
+        /// <summary>
+        /// Returns true if locals are to be initialized
+        /// </summary>
+        public virtual bool LocalsAreZeroed
```

Comment:
>public virtual bool LocalsAreZeroed [](start = 8, length = 35)

Adding virtual properties at the root is bug prone. We leant to always add abstract properties and have each derived type implement it and then have targeted tests to cover each implementation. I think you'd be better off with the cast in MethodCompiler, otherwise you will need to add and test implementations for retargeted methods, PE methods, etc. Does that worth it?

Reply:
Alternatively, we can explicitly throw for all symbols for which we don't  expect to call this API. Again this requires this method to have no default implementation.

Reply:
Can't we keep the virtual property and throw if the default implementation is called? Then only the method in the derived class could be used. I feel like it has the same result with less effort.

Reply:
The problem with having the property be virtual is future derivations of MethodSymbol. A year from now if a new derivation of MethodSymbol is added the developer may or may not know to re-evaluate the behavior of LocalsAreZeroed in the context of the type. Whether they do or don't comes down to the dillegence of the developer, and code reviewers, involved in the change. By making it abstract though we force the developer to think about this property and make a decision on its value.

Reply:
I guess, it is an option to throw from default implementation in this particular situation (for non public API used in very specific place and time in the pipeline), then, assuming that we have adequate test coverage, tests should catch derived classes that fail to override the method.

Reply:
In the end I set it to abstract. We're returning true in some derived classes for now (mostly because of synthesized members).